src/registers/control.rs

```diff
@@ -251,6 +251,28 @@ mod x86_64 {
             (frame, flags)
         }
 
+        /// Read the current P4 table address from the CR3 register along with PCID.
+        /// The correct functioning of this requires CR4.PCIDE = 1.
+        #[inline]
+        pub fn read_as_pcid() -> (PhysFrame, u16) {
+            let value: u64;
+
+            #[cfg(feature = "inline_asm")]
+            unsafe {
+                llvm_asm!("mov %cr3, $0" : "=r" (value));
+            }
+
+            #[cfg(not(feature = "inline_asm"))]
+            unsafe {
+                value = crate::asm::x86_64_asm_read_cr3();
+            }
```

Comment:
Instead of duplicating the `read` code here, we could introduce a new `CR3::read_raw() -> u64` function.